packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -21,12 +21,17 @@ import { MonacoEditorService } from './monaco-editor-service';
 import { MonacoQuickOpenService } from './monaco-quick-open-service';
 import { MonacoTextModelService } from './monaco-text-model-service';
 import { MonacoWorkspace } from './monaco-workspace';
+import { ThemeService } from '@theia/core/lib/browser/theming';
 
 import IEditorOverrideServices = monaco.editor.IEditorOverrideServices;
 
-const monacoTheme = 'vs-dark';
-monaco.editor.setTheme(monacoTheme);
-document.body.classList.add(monacoTheme);
+function changeTheme(editorTheme: string | undefined) {
+    const monacoTheme = editorTheme || 'vs-dark';
+    monaco.editor.setTheme(monacoTheme);
+    document.body.classList.add(monacoTheme);
+}
+changeTheme(ThemeService.get().getCurrentTheme().editorTheme);
```

Comment:
Could not it be done in the application contribution for the Monaco extension in `initialize` or  `initializeLayout`?

Reply:
see above

Reply:
ok